src/docks/timelinedock.cpp

```diff
@@ -680,23 +679,36 @@ void TimelineDock::selectMultitrack()
     emit selected(m_model.tractor());
 }
 
-void TimelineDock::copyClip(int trackIndex, int clipIndex)
+void TimelineDock::copyClip()
 {
-    if (trackIndex < 0)
-        trackIndex = currentTrack();
-    if (clipIndex < 0)
-        clipIndex = clipIndexAtPlayhead(trackIndex);
-    Q_ASSERT(trackIndex >= 0 && clipIndex >= 0);
-    QScopedPointer<Mlt::ClipInfo> info(getClipInfo(trackIndex, clipIndex));
-    if (info) {
-        QString xml = MLT.XML(info->producer);
-        Mlt::Producer p(MLT.profile(), "xml-string", xml.toUtf8().constData());
-        p.set_speed(0);
-        p.seek(info->frame_in);
-        p.set_in_and_out(info->frame_in, info->frame_out);
-        MLT.setSavedProducer(&p);
-        emit clipCopied();
+    if (selection().isEmpty())
+            return;
```

Comment:
This breaks the intentional behavior to act upon the topmost clip under the playhead.

Reply:
a fix: 256f1d8